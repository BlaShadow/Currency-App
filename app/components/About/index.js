import AboutComponent from './AboutComponent'
import Styles from './style'

export { AboutComponent, Styles } 