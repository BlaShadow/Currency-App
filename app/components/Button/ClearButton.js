import React from 'react';
import { View, TouchableOpacity, Text, Image } from 'react-native';
import color from 'color';

import styles from './style';

const ClearButton = ({onPress = () => {}, text = ""}) => {
    const highlightedButtonColor = color('white')
                                    .darken(0.4);
    
    return (
        <TouchableOpacity onPress={onPress}>
            <View style={styles.container}>
                <Image style={styles.icon} source={require('./Images/two_way.png')} />
                <Text style={styles.text}>{text}</Text>
            </View>
        </TouchableOpacity>
    );
}

export default ClearButton;