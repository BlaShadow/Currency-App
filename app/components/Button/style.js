import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
    container: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 10,
    },
    icon: {
        width: 20,
        height: 20,
        margin: 10
    }, 
    text: {
        fontSize: 14,
        color: 'white'
    }
});