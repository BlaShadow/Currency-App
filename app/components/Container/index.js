import Container, { Container as ContainerComponent } from './Container';
import styles from './styles';  

export { Container, styles, ContainerComponent };