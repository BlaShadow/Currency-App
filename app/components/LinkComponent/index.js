import { LinkComponent } from './LinkComponent'

export { LinkComponent }