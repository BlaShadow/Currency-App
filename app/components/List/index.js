import ListItem from './ListItem';
import Separator from './separator';

export { ListItem, Separator };