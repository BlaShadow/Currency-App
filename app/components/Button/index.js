import ClearButton from './ClearButton';
import styles from './style'; 

export {ClearButton, styles};