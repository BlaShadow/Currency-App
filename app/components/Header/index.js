import Header from './Header';
import style from './style';

export { Header, style };