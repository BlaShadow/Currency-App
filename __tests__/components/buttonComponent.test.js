import 'react-native';

import React from 'react'
import { ClearButton } from '../../app/components/Button'

import renderer from 'react-test-renderer';

test('Should render ClearButton correctly', () => {
    const tree = renderer.create(<ClearButton />).toJSON();

    expect(tree).toMatchSnapshot();
});