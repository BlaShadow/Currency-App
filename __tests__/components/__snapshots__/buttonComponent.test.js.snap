// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ClearButton correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={undefined}
  >
    <Image
      source={1}
      style={undefined}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
  </View>
</View>
`;
